tools/lib/html_branches.py

```diff
@@ -163,3 +190,21 @@ def html_tag_tree(text):
             stack.pop()
 
     return top_level
+
+
+def check_for_duplicate_ids(templates):
```

Comment:
This function doesn't actually check for duplicate ids; it just builds up a dictionary of them.  Please come up with a better name.